packages/components/pie-modal/src/index.ts

```diff
@@ -15,17 +17,19 @@ import {
 export { type ModalProps, headingLevels };
 
 const componentSelector = 'pie-modal';
+const validPropertyValues = validPropertyValuesDecoratorFactory(componentSelector);
+const requiredProperty = requiredPropertyDecoratorFactory(componentSelector);
 
 export class PieModal extends RtlMixin(LitElement) {
     @property({ type: Boolean })
         isOpen = false;
 
     @property({ type: String })
-    @requiredProperty(componentSelector)
+    @requiredProperty()
         heading!: string;
 
     @property()
-    @validPropertyValues(componentSelector, headingLevels, 'h2')
+    @validPropertyValues(headingLevels, 'h2')
```

Comment:
Similar here, what about `ValidValues` or `AllowedValues`?

Reply:
Similar to the other comment, when I wrote this initially I wasn't sure whether there might be other things that we need to decorate but I don't think that's the case now.

Reply:
Looking back we did have `validValues`, but there was a PR comment about potential confusion with its first parameter having the same name. I quite like `allowedValues`.